src/braket/pulse/waveforms.py

```diff
@@ -166,6 +185,25 @@ def sample(self, dt: float) -> np.ndarray:
         samples = self.iq * np.ones_like(sample_range)
         return samples
 
+    @staticmethod
+    def _from_json(waveform_json: Dict) -> ConstantWaveform:
+        wave_id = waveform_json["waveformId"]
+        length = iq = None
+        for val in waveform_json["arguments"]:
+            if val["name"] == "length":
+                length = (
+                    float(val["value"])
+                    if val["type"] == "float"
+                    else FreeParameterExpression(val["value"])
+                )
+            if val["name"] == "iq":
+                iq = (
+                    complex(val["value"])
+                    if val["type"] == "complex"
```

Comment:
- Is iq always supplied as type complex by the service? What if it's a real number and the service supplies float type? Would you end up casting to a FreeParameterExpression in that case? 
- Why do we use FreeParameterExpression instead of FreeParameter?

Reply:
* Yes, we will always get a complex number. In the case that it is a real number, the real component of the complex value will be present and the imaginary unit will be 0. Type is always expected to be complex even in this case.
* Here, expressions can be expected so this allows for that